sdk/testing/src/main/java/io/opentelemetry/sdk/testing/logs/TestLogRecordData.java

```diff
@@ -123,11 +138,24 @@ public Builder setObservedTimestamp(long timestamp, TimeUnit unit) {
 
     /** Set the body string. */
     public Builder setBody(String body) {
-      return setBody(Body.string(body));
+      return setAnyValueBody(AnyValue.of(body));
+    }
+
+    /**
+     * Set the body.
+     *
+     * @deprecated Use {@link #setAnyValueBody(AnyValue)}.
+     */
+    @Deprecated
+    public Builder setBody(io.opentelemetry.sdk.logs.data.Body body) {
+      if (body.getType() == io.opentelemetry.sdk.logs.data.Body.Type.STRING) {
+        setAnyValueBody(AnyValue.of(body.asString()));
+      }
```

Comment:
Omitted it because handling requires `setAnyValueBody` to accept a `@Nullable` AnyValue. `TestLogRecordData.Builder` instances start with a null `AnyValue`, so I figured a user who wants that should just avoid setting the body to something non-null, rather than setting to non-null and then later reseting to null.

It was probably a misguided attempt to avoid `@Nullable` though.